### PR TITLE
JoinEventForm fixes

### PR DESCRIPTION
**Pull request by hanswilw**
Based off @Hanse's #806 

- Fixes countdown bugs and handles the registration time in the provider
- Refactors to a HOC to avoid having to nest variables to keep the component somewhat clean
- Fixes stylings of buttons
image

### CODE REVIEW

**Comment by ekmartin** [comment]
nice button styling 😎

File: app/components/Form/Captcha.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Captcha extends Component<Props> {
           ref={ref => {
             this.captcha = ref;
           }}
-          sitekey={config.captchaKey}
+          sitekey={config.captchaKey || ''}
```
Comment (ekmartin) [inline app/components/Form/Captcha.js]: captchaKey has a default in master now

Comment (hanswilw) [reply app/components/Form/Captcha.js]: Right, didnt notice that it was added :)

**Comment by ekmartin** [comment]
I'll merge this @Hanse, but feel free to comment on it and we can fix stuff later.

**Pull request merged**